src/services/news/index.js

```diff
@@ -1,23 +1,246 @@
+
+class NewsService {
+	setup(app) {
+		this.app = app;
+	}
+
+	/**
+	 * Checks scoped permission for a user.
+	 * @param {BsonId|String} userId
+	 * @param {String} permission
+	 * @param {BsonId|String} targetId (optional) news target (scope) id. Only valid with targetModel.
+	 * @param {String} targetModel (optional) news target (scope) model. Only valid with target.
+	 * @returns {Promise<Boolean>} Promise that resolves to true/false
+	 * @example
+	 * await hasPermission(user._id, 'NEWS_VIEW') => true
+	 * await hasPermission(user._id, 'NEWS_CREATE', team._id, 'teams') => false
+	 * @memberof NewsService
+	 */
+	async hasPermission(userId, permission, targetId, targetModel) {
+		if (targetId && targetModel) {
+			const scope = this.app.service(`${targetModel}/:scopeId/userPermissions/`);
+			const params = { route: { scopeId: targetId.toString() } };
+			const scopePermissions = await scope.get(userId, params) || [];
+			return scopePermissions.includes(permission);
+		}
+		const user = await this.app.service('users').get(userId);
+		return user.permissions.includes(permission);
+	}
+
+	/**
+	 * Throw an error if the user is not allowed to perform the given operation
+	 * @param {News} news news item (required: {schoolId}, optional: {target, targetModel})
+	 * @param {Object} { userId, schoolId }
+	 * @param {String} permission permision to check
+	 * @returns undefined
+	 * @example
+	 * await authorize(news, params.account, 'NEWS_VIEW') => undefined
+	 * await authorize(news, params.account, 'NEWS_CREATE') => (throws Forbidden)
+	 * @memberof NewsService
+	 */
+	async authorize(news = {}, { userId, schoolId } = {}, permission) {
+		const authorized = await this.hasPermission(userId, permission, news.target, news.targetModel);
+		const sameSchool = news.schoolId.toString() === schoolId.toString();
+		if (!authorized || !sameSchool) {
+			throw new Forbidden('Not authorized.');
+		}
+	}
+
+	checkExistence(news, query) {
+		if (!news) {
+			logger.error(`Cannot find news item with query "${query}" => "${news}"`);
+			throw new NotFound('News item does not exist.');
+		}
+	}
+
+	/**
+	 * Returns all school news the user is allowed to see.
+	 * @param {Object} { userId, schoolId } -- The user's Id and schoolId
+	 * @returns Array<News Document>
+	 * @memberof NewsService
+	 */
+	async findSchoolNews({ userId, schoolId }) {
+		if (!this.hasPermission(userId, 'NEWS_VIEW')) {
+			throw new Forbidden('Mising permissions to view school news.');
+		}
+		return newsModel.find({	schoolId, target: { $exists: false } }).lean();
+	}
+
+	/**
+	 * Returns scoped news the user is allowed to see
+	 *
+	 * @param {BsonId|String} userId the user's Id
+	 * @param {BsonId|String} target (optional) Id of the news target (course, team, etc.)
+	 * @returns Array<News Document>
+	 * @memberof NewsService
+	 */
+	async findScopedNews(userId, target) {
+		const scopes = await newsModel.distinct('targetModel');
```

Comment:
note warum distinct

Reply:
`distinct('targetModel')` gibt alle verschiedenen Attributwerte für das Attribut `targetModel` in der News Collection zurück; also in diesem Fall alle (bisher in News verwendeten) Scopes